weather_sp/README.md

```diff
@@ -75,22 +73,45 @@ On the other hand, to specify a specific pattern use
 
 ## Output
 
-The base output path is specified using the `--output-path` flag. Based on that path, the directory structure of the
+The base output file names are specified using the `--output-template` flag. This template can be either a directory,
+or contain Python-style numbered formating.
+
+### Output template is a directory
+Based on the output template path, the directory structure of the
 input pattern is replicated.
 
 To create the directory structure, the common path of the input pattern is removed from the input file path and replaced
-with the output path.
+with the output path. A '_' is added to separate the split level / variable.
+
+Example:
+
+```bash
+--input-pattern 'gs://test-input/era5/2020/**' \
+--output-template 'gs://test-output/splits'
+```
+
+For a file `gs://test-input/era5/2020/02/01.nc` the output file pattern is
+`gs://test-output/splits/2020/02/01_` and if the temperature is a variable in that data, the output file for that
+split will be `gs://test-output/splits/2020/02/01_t.nc`
+
+### Output template with Python-style formating
+Using Python-style substitution (e.g. `{1}`) allows for more flexibility when creating the output files.
+The substitutions are based on the directory structure of the input file, where each `{<x>}` stands for
+one directory name, counting backwards from the end, i.e. the file name is `{0}`, the immediate
+directory in which it is located is `{1}`.
 
 Example:
 
 ```bash
 --input-pattern 'gs://test-input/era5/2020/**' \
---output-dir 'gs://test-output/splits'
+--output-template 'gs://test-output/splits/{2}.{1}.{0}T00.'
```

Comment:
I explained that above, the numbers count backwards from the filename. If there are a variable number of directories, then it depends what those directories are. I don't think there is another way of doing this that would avoid this issue if the directory structure is different for different files. I considered flipping the order and counting from the root level directory up with the filename having the highest number but decided against: for deeper directory hierarchies, usually the last couple of directories are what varies and is interesting in this case. It's also easier to count, if the directories of interest are a few levels down a 10+ level path it's easier to use {0}, {1}, {2} than to use {12},{13},{14} and still get it right. Also, if we flip and now the number of directories varies, there is a larger chance that we get numbers that are higher than the number of directories for a specific file. Alternatively,and maybe worse because harder to debug than a simple crash, we might lose the important parts, like the actual file name.

In summary, I don't think there is a way around this issue, and if the directory structure doesn't fit in a single template, it's probably not a good idea to use a single template, and rather use output-dir to keep the information that is encoded in those directories.

Reply:
I'm happy to document that users milage may vary based on their input pattern. One idea that comes to mind: Have the number of items in the output pattern depend on the format of the input pattern. For now, we can keep this approach in mind for another time.

Reply:
I think making this even more flexible requires more design. I also worry that we end up making this too error-prone that we accept all kinds of input and output pattern and create output files that may be unexpected. I think in some cases (like changing input directory depth) it may be better to fail loudly and then offer a flag for the user to override and run anyway. But I'd rather do that in a separate CL.